Formula/o/openjdk@17.rb

```diff
@@ -50,12 +50,13 @@ class OpenjdkAT17 < Formula
 
   fails_with gcc: "5"
 
-  # From https://jdk.java.net/archive/
+  # ARM64: https://www.azul.com/downloads/?version=java-17-lts&package=jdk
+  # Intel: https://jdk.java.net/archive/
   resource "boot-jdk" do
     on_macos do
       on_arm do
-        url "https://download.java.net/java/GA/jdk17.0.2/dfd4a8d0985749f896bed50d7138ee7f/8/GPL/openjdk-17.0.2_macos-aarch64_bin.tar.gz"
-        sha256 "602d7de72526368bb3f80d95c4427696ea639d2e0cc40455f53ff0bbb18c27c8"
+        url "https://cdn.azul.com/zulu/bin/zulu17.46.19-ca-jdk17.0.9-macosx_aarch64.tar.gz"
+        sha256 "d6837676e55b97772b6512e253fdaf8ab282bb216c0f8366b6c5905cd02b5056"
```

Comment:
Was there a specific issue that required updating JDK on ARM?

Reply:
Same major version may be worth it here as there is no OpenJDK 16 ARM release. It would allow avoiding a 3rd party JDK. For older OpenJDK, it may depend on if there are any required fixes, e.g. OpenJDK 10 is no longer supported so may not get all backports (Mainly issue for new platforms like Apple ARM).

Unless there is a required fix, I think we can stick to OpenJDK's binaries for bootstrap on newer OpenJDK formulae (`openjdk@17` and `openjdk`).

Reply:
For OpenJDK, `n-1` refers to previous major version and is the standard choice of boot JDK.

Changing patch release usually has minor impact and OpenJDK doesn't provide them after a certain point. They point to Temurin releases instead https://wiki.openjdk.org/display/JDKUpdates/JDK+17u#JDK17u-Downloads

Reply:
just a bootstrap jdk, it is fine to use whatever latest

Reply:
> just a bootstrap jdk, it is fine to use whatever latest

The main thing is we only resort to Zulu as a last resort (e.g. OpenJDK7 doesn't exist and OpenJDK11 doesn't provide ARM release).

Reply:
Anyway, I think this should be reverted unless there is a specific issue. And if there is a specific issue, it should be included in the comment.

For example, the reason we use Zulu on `openjdk@8`:
https://github.com/Homebrew/homebrew-core/blob/bceb694c098acc96adab7b05b0a5b32b4332add9/Formula/o/openjdk%408.rb#L50-L52

Reply:
We definitely should not use a random non-open boot JDK here.